sklearn/metrics/tests/test_classification.py

```diff
@@ -398,6 +427,60 @@ def test_matthews_corrcoef():
                   matthews_corrcoef(y_1, y_2, sample_weight=mask), 0.)
 
 
+def test_matthews_corrcoef_multiclass():
+    rng = np.random.RandomState(0)
+    ord_a = ord('a')
+    n_classes = 4
+    y_true = [chr(ord_a + i) for i in rng.randint(0, n_classes, size=20)]
+
+    # corrcoef of same vectors must be 1
+    assert_almost_equal(matthews_corrcoef(y_true, y_true), 1.0)
+
+    # with multiclass > 2 it is not possible to achieve -1
+    y_true = [0, 0, 1, 1, 2, 2]
+    y_pred_bad = [2, 2, 0, 0, 1, 1]
+    assert_almost_equal(matthews_corrcoef(y_true, y_pred_bad), -.5)
+
+    # Maximizing false positives and negatives minimizes the MCC
+    # The minimum will be different for depending on the input
+    y_true = [0, 0, 1, 1, 2, 2]
+    y_pred_min = [1, 1, 0, 0, 0, 0]
+    assert_almost_equal(matthews_corrcoef(y_true, y_pred_min), -0.6123724)
```

Comment:
This is simply the correct output for this specific multiclass instance. The reason why this specific example takes a weird value and not -1 is because technically some of the negative predictions are correct. When there are 2 classes you can construct an instance that is completely wrong, but in more than 2 classes every time you say class 2 when it should have been class 1, you are technically correct that it wasn't class 0, so you'll always get something right. 

Perhaps `-12 / np.sqrt(24 * 16)` would be better? I'm not sure how to give a better intuition without redefining the function itself or using a lot of terms. 

I actually found this particular example by doing a brute force search over 6 examples with 3 labels to find the minimum value the MCC would take in this instance.